src/taskEvents/script.js

```diff
@@ -133,14 +147,9 @@ function closeGenericModal() {
   containerDiv.innerHTML = '';
 }
 
-function removeSkill(e) {
-  // this is not how it's going to be done, when we have the userSkills this function is going to change
-  e.target.parentElement.remove();
-}
-
-function openAddSkillModal() {
-  const skills = ['frontend', 'backend', 'System Design']; // this is temporary data that will be removed once we have userSkill collection in our DB
-  const level = [1, 2, 3, 4, 5];
+function openAddSkillModal(userid, rolesDiv) {
+  const skills = tagLevelOptions.allTags;
```

Comment:
Please make sure that `tagLevelOptions` has data before this function executes.

Reply:
Also, let's try to improve the function placement in this file as it has become difficult to understand the flow. Maybe we can create a separate issue for it
Reference - https://javascript.info/coding-style#function-placement

Reply:
thanks for the reference, I can change the function placement in this PR